### PR TITLE
feat: implement textarea, masked fields and spinner fields

### LINKED ISSUE #192: Create getter for Expanse in NumericBox, remove Theme

### PR DESCRIPTION
**Pull request by hyyan**
The PR adds the following new components:

1. `MaskedDateField`
2. `MaskedDateFieldSpinner`
3. `MaskedNumberField`
4. `MaskedNumberFieldSpinner`
5. `MaskedTimeField`
6. `MaskedTimeFieldSpinner`
7. `MaskedTextField`
8. `MaskedTextFieldSpinner`
9. `TextArea`
10. `TextFieldSpinner`

Additionally, the PR introduces `MaskDecorator` as a foundation for parsing and masking according to BBj-defined rules.

closes #192 , #173, #172, #170, #169 , #165 , #162, #103, #101 , #89 , #61

### CODE REVIEW

**Comment by hyyan** [comment]
@MatthewHawkins please go over the javadocs and fix any obvious issues.

File: webforj-foundation/src/main/java/com/webforj/component/field/MaskedDateField.java [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+ * <td>d</td>
+ * <td>Decimal (default format)</td>
+ * </tr>
+ * </table>
+ *
+ * <h3>Date Parser</h3>
+ *
+ * <p>
+ * The date parser interprets date strings based on the date mask. For example, if the date mask is
+ * %Mz/%Dz/%Yz (US format, the BBj default setting), the date is parsed with the understanding that
```
Comment (MatthewHawkins) [inline webforj-foundation/src/main/java/com/webforj/component/field/MaskedDateField.java:100]: Leave mention of BBj here?

Comment (hyyan) [reply webforj-foundation/src/main/java/com/webforj/component/field/MaskedDateField.java:100]: Should be webforJ

**Review by MatthewHawkins — APPROVED** [review]
Javadoc looks good!

**Pull request merged**